pkg/util/web3/eip55.go

```diff
@@ -0,0 +1,54 @@
+package web3
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/authgear/authgear-server/pkg/util/hexstring"
+	"golang.org/x/crypto/sha3"
+)
+
+// https://eips.ethereum.org/EIPS/eip-55
+type EIP55 string
+
+var parseRegExp = regexp.MustCompile(`^(0x)0*([0-9a-fA-F]+)$`)
+
+func NewEIP55(s string) (EIP55, error) {
+	if !parseRegExp.MatchString(s) {
+		return "", fmt.Errorf("hex string must match the regexp %q", parseRegExp)
+	}
+
+	strippedHex := strings.ToLower(strings.ReplaceAll(s, "0x", ""))
+
+	sha := sha3.NewLegacyKeccak256()
```

Comment:
Can we use go-ethereum to help us implement the parsing? We still want to keep `type EIP55 string`, only use go-ethererum for the implementation.